extensions/amp-story/1.0/amp-story.js

```diff
@@ -2143,13 +2144,16 @@ export class AmpStory extends AMP.BaseElement {
       return false;
     }
 
-    pageBeforeEl.setAttribute(Attributes.ADVANCE_TO, pageToBeInsertedId);
+    const advanceAttr = isExperimentOn(this.win, 'amp-story-branching') ?
+      Attributes.PUBLIC_ADVANCE_TO : Attributes.ADVANCE_TO;
+
+    pageBeforeEl.setAttribute(advanceAttr, pageToBeInsertedId);
     pageBeforeEl.setAttribute(Attributes.AUTO_ADVANCE_TO, pageToBeInsertedId);
     pageToBeInsertedEl.setAttribute(Attributes.RETURN_TO, pageBeforeId);
 
     const nextPageEl = nextPage.element;
     const nextPageId = nextPageEl.id;
-    pageToBeInsertedEl.setAttribute(Attributes.ADVANCE_TO, nextPageId);
+    pageToBeInsertedEl.setAttribute(Attributes.advanceAttr, nextPageId);
```

Comment:
Should we also change the one in the `next_` method for this PR to work?

Reply:
Thank you for catching this! I never caught it in testing because this check addresses the case where the last page has the private attribute. 

I'm assuming that the check for the private attribute was originally added in the case where an ad should follow the last page. I made the change/added the check, but I'm also not sure what the use case for allowing the last page of a story to have a public advance-to attribute would be. It seems like it would more likely than not create an infinite cycle since a story would be able to 'end'.

Reply:
I am a bit confused here. This is the method that `amp-story-auto-ads` calls when it wants to add a new page to the story. So if we have a story like `page1 -> page2` we can change it to  `page1 -> ad -> page2`. All this logic is just trying to copy any pointers from page1 to the ad so that the flow is the same.

Reply:
I see, I think that the the idea there was just future proofing knowing that this eventually may be used for non linear stories, but @gmajoulet is right, as it is written now it won't respect the experiment.

Reply:
(the ad can't be the last page of a story anyway) https://github.com/ampproject/amphtml/blob/5ce6f620b20147a4492491e9781527d08257a294/extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js#L682-L684